.github/workflows/up-spec-badge.yml

```diff
@@ -0,0 +1,95 @@
+name: Update up-spec compatibility badge
+
+on:
+  push:
+    branches: [ main ]
```

Comment:
My understanding is that the whole mechanism will (currently) only work when we make changes to the main branch. However, IMHO we will need to be able to do this on (version specific) branches as well. For example once we have released stable versions 1.0.0 and 2.0.0 which implement different versions of the spec, we will need to be able to apply this mechanism on version specific bug-fix branches. AFAICT this mechanism should generally also work in such a scenario, but will need to consider the branch that we are pushing to in several places, or am I mistaken?